### ISSUE TITLE
nf() not rounding correctly

### ISSUE THREAD

**Issue by ChristerNilsson** [processing/p5.js#5959]
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [X] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build Process
- [ ] Unit Testing
- [ ] Internalization
- [ ] Friendly Errors
- [ ] Other (specify if possible)

### p5.js version

v1.5.0 October 18, 2022

### Web browser and version

Chrome: 109.0.5414.74

### Operating System

W10 Home 19045.2486

### Steps to reproduce this

```js
console.log(Math.PI)
console.log(nf(Math.PI,1,4))

PI is approx   3.141592653589793
expect answer: 3.1416
actual answer: 3.1415
```
It seems a simple truncation is used instead of correct rounding.

**Comment by davepagurek** [comment]
Thanks @akp660! I'll assign this to you.

**Comment by Qianqianye** [comment]
Just link another nf() related issue here #5710: nf() produces problematic string-formatting of very large or small numbers.

**Comment by snehilchhabria** [comment]
Is this issue resolved already? I may be able to help if no one's working on this.

**Comment by Qianqianye** [comment]
> Is this issue resolved already? I may be able to help if no one's working on this.

Thanks @snehilchhabria for offering help. @akp660 has already been assigned to this issue and is in the process of resolving it in this pull request #5973.

**Comment by snehilchhabria** [comment]
okay thanks for your reply.

**Comment by owais0017** [comment]
hey i want to work on this issue. where do i have to make the changes?

**Comment by Qianqianye** [comment]
Thanks @owais0017  for offering help. @aditya-shrivastavv is in the process of resolving it in this pull request https://github.com/processing/p5.js/pull/5973. Please feel free to take a look at other open issues here. Thank you!

**Comment by ChristerNilsson** [comment]
Normal procedure is at least to fork the project involved. After six weeks I havent seen this happen. Maybe time to let someone more eager try.
There are two candidates above.

**Comment by limzykenneth** [comment]
Closed with #5973
